libc/src/stdlib/qsort_r.cpp

```diff
@@ -19,13 +19,12 @@ LLVM_LIBC_FUNCTION(void, qsort_r,
                    (void *array, size_t array_size, size_t elem_size,
                     int (*compare)(const void *, const void *, void *),
                     void *arg)) {
-  if (array == nullptr || array_size == 0 || elem_size == 0)
-    return;
-  internal::Comparator c(compare, arg);
-  auto arr = internal::Array(reinterpret_cast<uint8_t *>(array), array_size,
-                             elem_size, c);
 
-  internal::sort(arr);
+  internal::unstable_sort(
+      array, array_size, elem_size,
+      [compare, arg](const void *a, const void *b) noexcept -> bool {
```

Comment:
@michaelrj-google I think the `is_less` function @Voultapher passed in is actually a wrapper of the raw integer function. So, calling it `is_less` should be good.